### PR TITLE
mongodb-3.x-plugin: add support for previous versions of 3.3

### PR DESCRIPTION
**Pull request by SataQiu**
Please answer these questions before submitting pull request

- Why submit this pull request?
- [x] Bug fix
- [ ] New feature provided
- [ ] Improve performance

- Related issues #2738 

___
### Bug fix
- Bug description.

- How to fix?

___
### New feature or improvement
- Describe the details and related test reports.

### CODE REVIEW

File: apm-sniffer/apm-sdk-plugin/mongodb-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/mongodb/v3/MongoDBMethodInterceptor.java [hunk 1]
```diff
@@ -201,7 +201,13 @@ private String limitFilter(String filter) {
     public void onConstruct(EnhancedInstance objInst, Object[] allArguments) {
         Cluster cluster = (Cluster)allArguments[0];
         StringBuilder peers = new StringBuilder();
-        for (ServerDescription description : cluster.getDescription().getServerDescriptions()) {
+        Collection<ServerDescription> descriptions;
+        try {
+            descriptions = cluster.getDescription().getServerDescriptions();
+        } catch (Throwable t) {
+            descriptions = cluster.getDescription().getAll();
```
Comment (wu-sheng) [inline apm-sniffer/apm-sdk-plugin/mongodb-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/mongodb/v3/MongoDBMethodInterceptor.java]: Please use witness class to identify versions, using exception is very bad for performance.

**Comment by wu-sheng** [comment]
Please follow the suggestion. We have provided a better way.

File: apm-sniffer/apm-sdk-plugin/mongodb-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/mongodb/v3/MongoDBMethodInterceptor.java (hunk 1 above)
Comment (SataQiu) [reply apm-sniffer/apm-sdk-plugin/mongodb-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/mongodb/v3/MongoDBMethodInterceptor.java]: Thanks for your feeedback @wu-sheng 
I think I need to refer to some current code about witness class.
Any example code for using witness class?

Comment (wu-sheng) [reply apm-sniffer/apm-sdk-plugin/mongodb-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/mongodb/v3/MongoDBMethodInterceptor.java]: Witness class is very easy. Take `MongoDBCollectionInstrumentation` as example. 
Witness class is using a special class name in this version to tell agent core, only active the instrumentation when it exists.

Comment (SataQiu) [reply apm-sniffer/apm-sdk-plugin/mongodb-3.x-plugin/src/main/java/org/apache/skywalking/apm/plugin/mongodb/v3/MongoDBMethodInterceptor.java]: Thanks a lot :+1: 
I will try to use witeness class.

**Pull request closed without merging**